src/awkward/_v2/contents/recordarray.py

```diff
@@ -315,7 +315,11 @@ def _getitem_field(self, where, only_fields=()):
                 return self.content(where)._getitem_fields(nexthead, nexttail)
 
     def _getitem_fields(self, where, only_fields=()):
-        indexes = [self.field_to_index(field) for field in where]
+        if len(where) == 0:
+            indexes = range(len(self._contents))
+        else:
+            indexes = [self.field_to_index(field) for field in where]
```

Comment:
```suggestion
            indexes = [self.field_to_index(field) for field in where]
            if self._fields is None:
                fields = None
            else:
                if len(where) == 0:
                    fields = [self._fields[i] for i in range(len(self._contents))]
                else:
                    fields = [self._fields[i] for i in indexes]
```

In here `indexes` should not be changed, because they are used later on (L333+) to pick up correct array elements accordingly. 

In your above changes, this test will fail: 

`assert ak._v2.to_list(ak._v2.Array([{"x": [[1, 2, 3], [4, 5, 6]]}])[:, []]) == [{}]` 

Because the slice will output the following:

`ak._v2.to_list(ak._v2.Array([{"x": [[1, 2, 3], [4, 5, 6]]}])[:, []])
[{'x': [[1, 2, 3], [4, 5, 6]]}]`


However, this will mean that when slicing with empty arrays `len(fields)` will be > 0 while `len(contents) == 0` and will cause a `ValueError` at L96 which can be fixed with `if len(contents) > 0 and not len(contents) == len(fields):` (in the issue, @jpivarski pointed out that even if the length is 0, the field should be passed on). 

Moreover, accepting different lengths will require another fix in File "/awkward/_v2/highlevel.py", line 1175, in __repr__ , into
`return self._contents[index] IndexError: list index out of range`

Reply:
Ah, thanks for this. I've misunderstood the intent of Jim's test case. The empty slice should still drop the contents, but keep the fields (whereas I was leaving the contents for subsequent slicing (that never happens)).

Reply:
It sounds like this is resolved, right?

(It's complicated and I'm just catching up.)

Reply:
Yes, it is. I misunderstood the problem at this point.